### PR TITLE
Feature/dependency update

### PR DESCRIPTION
**Pull request by dankelleher**
Update dependencies, including lint upgrades.
Refactored constraint evaluation
Cleaned up some tests (removed duplicated code)

*Fixed AND logic when validating constrants.*

https://github.com/identity-com/credential-commons/pull/146/files#diff-33f53056d40cd4232bb0d8a33a257706102a38fd29b8b22059d10dbc9ca9458bL698

I removed this line, which always overwrites the previous "result" value with the current one, when looping through constraints, as it seemed to make only the last constraint relevant. i.e. if you had three constraints, that resolved to [false, false, true], the entire set would resolve to true. I cleaned up this code in general so now it ANDs correctly I think, and added a test

*Defaulting to false when verifying a temporary attestation:*

https://github.com/identity-com/credential-commons/pull/146/files#diff-33f53056d40cd4232bb0d8a33a257706102a38fd29b8b22059d10dbc9ca9458bR575

There was a test that applied to this which I don't think ever worked - it returned a false positive due to the try-catch. I am not sure if SIP etc actually call verify on temporary attestations but this would trigger unnecessary calls in the IDV. I fixed the test and added this logic.

### CODE REVIEW

File: src/creds/VerifiableCredential.js [hunk 1]
```diff
@@ -801,7 +725,7 @@ function transformMetaConstraint(constraintsMeta) {
 
 /**
  * isMatchCredentialMeta
```
Comment (kevinhcolgan) [inline src/creds/VerifiableCredential.js:727]: should probably have an explanation of what this function is doing as it's non-obvious

File: src/creds/VerifiableCredential.js [hunk 2]
```diff
@@ -768,7 +692,7 @@ const CREDENTIAL_META_FIELDS = [
  *
  * @param {*} vc
```
Comment (kevinhcolgan) [inline src/creds/VerifiableCredential.js:693]: missing @return value in jsdoc

File: src/creds/VerifiableCredential.js [hunk 3]
```diff
@@ -812,12 +736,16 @@ function transformMetaConstraint(constraintsMeta) {
  * //   }
  */
```
Comment (kevinhcolgan) [inline src/creds/VerifiableCredential.js:737]: missing @return JSDoc

File: src/creds/VerifiableCredential.js [hunk 4]
```diff
@@ -1,51 +1,35 @@
...
 const time = require('../timeHelper');
+const { CvcMerkleProof } = require('./CvcMerkleProof');
+const { ClaimModel } = require('./ClaimModel');
 
-function sha256(string) {
-  return sjcl.codec.hex.fromBits(sjcl.hash.sha256.hash(string));
-}
-
-const convertTimestamp = delta => time.applyDeltaToDate(delta).getTime() / 1000;
+const convertTimestamp = (delta) => time.applyDeltaToDate(delta).getTime() / 1000;
```
Comment (kevinhcolgan) [inline src/creds/VerifiableCredential.js:20]: A JSDoc would be good for this fn. Convert to what?

**Review by kevinhcolgan — APPROVED** [review]

File: src/creds/VerifiableCredential.js [hunk 5]
```diff
@@ -681,24 +603,28 @@ function VerifiableCredentialBaseConstructor(identifier, issuer, expiryIn, ucas,
...
       const pathValue = _.get(claims, path);
       if (isDateStructure(pathValue)) {
         _.set(claims, path, transformDate(pathValue));
         // transforms delta values like "-18y" to a proper timestamp
-        // eslint-disable-next-line no-confusing-arrow
-        _.set(constraint, path, _.mapValues(constraint[path], obj => _.isString(obj) ? convertTimestamp(obj) : obj));
-        // _.set(constraint, path, _.mapValues(constraint[path], obj => _.isString(obj) ? timestamp.now(obj) : obj));
+        _.set(constraint, path, _.mapValues(constraint[path], convertTimestampIfString));
       }
-      result = (sift.indexOf(constraint, [claims]) > -1);
```
Comment (dankelleher) [inline src/creds/VerifiableCredential.js:698]: @jpsantosbh  here (1)

File: src/creds/VerifiableCredential.js [hunk 6]
```diff
@@ -652,9 +567,16 @@ function VerifiableCredentialBaseConstructor(identifier, issuer, expiryIn, ucas,
    */
   this.verifyAttestation = async () => {
     // Don't check attestation for credentials that are never attested on blockchain
-    if (this.proof.anchor.type === 'transient' || this.proof.anchor.network === 'dummynet') {
+    if (
+      this.proof.anchor.type === 'transient' || this.proof.anchor.network === 'dummynet') {
       return true;
     }
+
+    if (
```
Comment (dankelleher) [inline src/creds/VerifiableCredential.js:575]: @jpsantosbh here (2)

File: __test__/creds/VerifiableCredential.test.js [hunk 7]
```diff
@@ -1071,9 +1045,21 @@ describe('Unit tests for Verifiable Credentials', () => {
     })).toBeTruthy();
   });
 
+  it('Should fail with two constraints if one of them fails', () => {
```
Comment (dankelleher) [inline __test__/creds/VerifiableCredential.test.js:1048]: @jpsantosbh test for (1)

File: __test__/creds/VerifiableCredential.test.js [hunk 8]
```diff
@@ -996,35 +974,32 @@ describe('Unit tests for Verifiable Credentials', () => {
...
+
+    cred.proof.anchor.network = 'mainnet';
+
     const validation = await cred.verifyAttestation();
     expect(validation).toBeFalsy();
     done();
   });
 
-  it('should fail the check with temporary attestations faked as permanent', async (done) => {
+  it('should fail the check with temporary attestations faked as permanent', async () => {
```
Comment (dankelleher) [inline __test__/creds/VerifiableCredential.test.js:989]: @jpsantosbh test for (2)

**Pull request merged**